### ISSUE TITLE
@solidjs/lazy breaks HMR

### ISSUE THREAD

**Issue by 14gasher** [withastro/astro#6886]
### What version of `astro` are you using?

2.3.0

### Are you using an SSR adapter? If so, which one?

Node

### What package manager are you using?

yarn

### What operating system are you using?

Mac

### What browser are you using?

Safari

### Describe the Bug

Using lazy from solid breaks HMR. After making a change to one of the route component files, the entire route just unmounts and disappears haha. There is no state or anything.

Removing the lazy components brings the HMR back correctly.

See the example and try editing the C2 file after navigating to the app/c2 route. Everything will disappear, and you will need to click back to c1, then back to c2.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-x9meuo?file=src/App/C2.tsx

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by natemoo-re** [comment]
Going to close this as a duplicate of https://github.com/withastro/astro/issues/6886. Astro currently doesn't support Solid's `lazy` construct, but there's work in https://github.com/withastro/astro/pull/6791 to unblock this!
